services/workflows-service/src/collection-flow/collection-flow.service.ts

```diff
@@ -229,6 +238,45 @@ export class CollectionFlowService {
     return flowData;
   }
 
+  private async __persistFileUrlsToDocuments(
+    documents: TDocument[] = [],
+    projectIds: TProjectIds,
+  ): Promise<TDocument[]> {
+    const fileEntities = await Promise.all(
+      documents.reduce((filesList, document) => {
+        document.pages.forEach((page: { ballerineFileId: string }) => {
+          filesList.push(this.storageService.getFileById({ id: page.ballerineFileId }, projectIds));
```

Comment:
TypeScript suggests `ballerineFileId` may be `undefined`. We should consider handling that case.